sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/FullFidelityChangeFeedProcessorBuilder.java

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.changefeed.fullfidelity.ChangeFeedProcessorBuilderImpl;
+import com.azure.cosmos.models.ChangeFeedProcessorOptions;
+import com.azure.cosmos.models.ChangeFeedProcessorItem;
+import com.azure.cosmos.util.Beta;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * Helper class to build a {@link ChangeFeedProcessor} instance.
+ *
+ * <!-- src_embed com.azure.cosmos.fullFidelityChangeFeedProcessor.builder -->
+ * <pre>
+ * ChangeFeedProcessor changeFeedProcessor = new FullFidelityChangeFeedProcessorBuilder&#40;&#41;
+ *     .hostName&#40;hostName&#41;
+ *     .feedContainer&#40;feedContainer&#41;
+ *     .leaseContainer&#40;leaseContainer&#41;
+ *     .handleChanges&#40;docs -&gt; &#123;
+ *         for &#40;JsonNode item : docs&#41; &#123;
+ *             &#47;&#47; Implementation for handling and processing of each JsonNode item goes here
+ *         &#125;
+ *     &#125;&#41;
+ *     .buildChangeFeedProcessor&#40;&#41;;
+ * </pre>
+ * <!-- end com.azure.cosmos.fullFidelityChangeFeedProcessor.builder -->
+ */
+@Beta(value = Beta.SinceVersion.V4_34_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+public class FullFidelityChangeFeedProcessorBuilder {
```

Comment:
thinking in the future of how we want to support merge (starting with epkRange based lease only), I feel like it makes more sense to me by using ChangeFeedProcessorV2Builder + incrementalMode()/fullFidelityMode()

Reply:
+1

Reply:
To use handleChanges overload + mode APIs on the same builder. 
throw errors on incompatible mode + handle API.